src/containers/GrantsTableWrapper.js

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+
+import { uniq, map, filter, findIndex, sortBy } from 'lodash';
+
+import { useQuery, gql } from '@apollo/client';
+
+import { stripHtml, extractYear } from '../utils';
+
+import GrantTable from '../components/GrantTable';
+
+const GET_ORGANIZATION_GRANTS = gql`
+  query getOrg($organizationId: String!, $grantsFundedOffset: Int, $grantsReceivedOffset: Int, $limit: Int) {
+    organization(uuid: $organizationId) {
+      grantsFunded (offset: $grantsFundedOffset, limit: $limit, orderBy: uuid) {
+        uuid
+        dateFrom
+        dateTo
+        to {
+          name
+          uuid
+        }
+        amount
+        description
+      }
+      grantsReceived (offset: $grantsReceivedOffset, limit: $limit, orderBy: uuid) {
+        uuid
+        dateFrom
+        dateTo
+        from {
+          name
+          uuid
+        }
+        amount
+        description
+      }
+    }
+  }
+`;
+
+const GrantsTableWrapper = (props) => {
+  const { loading, error, data, fetchMore } = useQuery(GET_ORGANIZATION_GRANTS, {
+    variables: { 
+      organizationId: props.organizationId,
+      grantsFundedOffset: 0,
+      grantsReceivedOffset: 0,
+      limit: 100
+    },
+    notifyOnNetworkStatusChange: true,  // Update 'loading' prop after fetchMore
+  });
+
+  if (loading) return 'Loading...';
+  if (error) return `Error! ${error}`;
+
+  if (!data.organization) return `Failed to load org data!`;
+
+  if (data.organization.grantsFunded.length < props.countGrantsFrom || data.organization.grantsReceived.length < props.countGrantsTo) {
+    fetchMore({
+      variables: {
+        grantsFundedOffset: data.organization.grantsFunded.length,
+        grantsReceivedOffset: data.organization.grantsReceived.length,
+      },
+    });
+  }
+
+  console.log(
+    `Fetched ${data.organization.grantsFunded.length}/${props.countGrantsFrom} grants funded\
+     and ${data.organization.grantsReceived.length}/${props.countGrantsTo} grants received.`
+  );
+
+  const {
```

Comment:
Moving this block up around line 55 would shorten some of the code above

Reply:
Ah so this is a bit tricky -- we need the orginal length of the grantsFunded/grantsReceived response pre-cleanse in order to properly set offset. Because the cleanse creates the table "org header rows" and inserts them directly into the array, we'd be trying to fetch something like 182/171 possible grants if this is defined upfront.